packages/calypso-e2e/tsconfig.json

```diff
@@ -1,10 +1,9 @@
 {
 	"extends": "@automattic/calypso-build/typescript/ts-package.json",
 	"compilerOptions": {
-		"module": "commonjs",
-		"outDir": "dist/cjs",
+		"outDir": "dist/esm",
 		"declarationDir": "dist/types",
 		"rootDir": "."
 	},
-	"include": [ "src", "test" ]
+	"include": [ "src" ]
```

Comment:
This will exclude `test/*` files from type checking, right?

Reply:
It does, and I just confirmed that unfortunately when jest uses babel, it does no type checking. I spun up a separate test-only tsconfig and updated the `package.json` scripts, so now you can run `yarn test` and it will do a quick no-emit type check and then run the unit tests!

Reply:
Type-checking is already done by a TeamCity job that does ` tsc --build packages/*/tsconfig.json` (https://github.com/Automattic/wp-calypso/blob/trunk/.teamcity/_self/projects/WebApp.kt#L533).

Having `test` here is mostly to help VSCode provide smart language features.

I think removing `test` will break those two systems. What was the reason to drop it?

Reply:
> I think removing test will break those two systems. What was the reason to drop it?

There's not a lot of urgent downsides here, it just seemed strange to include the unit test files in the package output build that would be consumed by other parts of the monorepo. It increases that package build size and consumers could see confusing or unneeded exports if any of the unit tests ended up exporting anything (or more realistically, a shared utility or helper file used by the unit tests).

It looks like this is the same pattern taken by other packages in the monorepo (e.g. `explat-client` and `i8n-utils`), except they don't add anything that does any type checking for the unit tests.

I definitely can roll this back to having `test` in the `tsconfig.json` for now if that's easiest! I was just looking to see if there's a way to avoid adding things to the package build output that are not intended for consumers and only for the package developers. There's definitely a lot of different ways we could go about that too!